exporters/ostream/src/log_exporter.cc

```diff
@@ -0,0 +1,92 @@
+#include "opentelemetry/exporters/ostream/log_exporter.h"
+
+#include <iostream>
+
+namespace nostd   = opentelemetry::nostd;
+namespace sdklogs = opentelemetry::sdk::logs;
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace exporter
+{
+namespace logs
+{
+OStreamLogExporter::OStreamLogExporter(std::ostream &sout) noexcept : sout_(sout) {}
+
+sdklogs::ExportResult OStreamLogExporter::Export(
+    const nostd::span<std::shared_ptr<opentelemetry::logs::LogRecord>> &records) noexcept
+{
+  if (isShutdown_)
+  {
+    return sdklogs::ExportResult::kFailure;
+  }
+
+  for (auto &record : records)
+  {
+    // Convert trace, spanid, traceflags into string convertable representation
+    constexpr int trace_id_bytes      = 16;
+    char trace_id[trace_id_bytes * 2] = {0};
+    record->trace_id.ToLowerBase16(trace_id);
+
+    constexpr int span_id_bytes     = 8;
+    char span_id[span_id_bytes * 2] = {0};
+    record->span_id.ToLowerBase16(span_id);
+
+    constexpr int trace_flags_bytes         = 1;
+    char trace_flags[trace_flags_bytes * 2] = {0};
+    record->trace_flags.ToLowerBase16(trace_flags);
+
+    /*** Print out each field of the log record ***/
+
+    // Print fields most useful to user first
+    sout_ << "{\n"
+          << "    timestamp   : " << record->timestamp.time_since_epoch().count() << "\n"
+          << "    severity    : " << severityNumToText[static_cast<int>(record->severity)] << "\n"
```

Comment:
What guarantees that record->severity won't be out of bounds?

Reply:
I'm thinking that the only way this would be out of bounds is if the user manually sets the severity to a value larger than 24, which would be difficult for them to do since a [class enum](https://github.com/open-telemetry/opentelemetry-cpp/blob/master/api/include/opentelemetry/logs/log_record.h#L39) doesn't let the user implicitly cast from int.